### PR TITLE
feat(ranger-plugins): enable log4j for auditlogs

### LINKED ISSUE #833: Enable sending ranger auditlogs to filesystem
New feature request: Add an option to enable ranger audit logs to be sent to filesystem

### PR DESCRIPTION
**Pull request by giem-git**
#### Which issue(s) this PR fixes

Fixes #833 

#### Agreements

- [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: roles/hbase/master/tasks/config.yml [hunk 1]
```diff
@@ -27,7 +27,8 @@
     owner: root
     group: root
     mode: "644"
-
+  vars:
+    ranger_audit_file: "{{ hbase_log_dir }}/{{ hbase_master_ranger_audit_file }}"
 - name: Render hbase-site.xml
```
[the remark this answers is not in the document]
Comment (giem-git) [reply roles/hbase/master/tasks/config.yml:33]: fixed in last push

File: roles/hive/common/templates/hive-log4j2.properties.j2 [hunk 2]
```diff
@@ -62,8 +63,31 @@ appender.DRFA.policies.timebased.modulate = false
 appender.DRFA.strategy.type = DefaultRolloverStrategy
 appender.DRFA.strategy.max = {{ hive_log_drfa_maxbackupindex }}
 
+{% if enable_ranger_audit_log4j %}
+appender.RANGERAUDIT.type=RollingRandomAccessFile
+appender.RANGERAUDIT.name=RANGERAUDIT
+appender.RANGERAUDIT.fileName=${sys:hive.log.dir}/{{ hive_ranger_audit_file }}
```
[the remark this answers is not in the document]
Comment (giem-git) [reply roles/hive/common/templates/hive-log4j2.properties.j2]: fixed in last push

File: roles/hive/common/templates/hive-log4j2.properties.j2 [hunk 3]
```diff
@@ -26,7 +26,8 @@ property.hive.log.file = hive.log
 property.hive.perflogger.log.level = INFO
 
 # list of all appenders
-appenders = {{ hive_root_logger }}
+appenders = {%- if enable_ranger_audit_log4j %}RANGERAUDIT, {% endif %}{{ hive_root_logger }}
```
[the remark this answers is not in the document]
Comment (kpgtek) [reply roles/hive/common/templates/hive-log4j2.properties.j2:29]: I think the space character is removed by the ```-``` character in ```{%- if ... }```.

[the remark this answers is not in the document]
Comment (giem-git) [reply roles/hive/common/templates/hive-log4j2.properties.j2]: Fixed in last push

**Review by kpgtek — APPROVED** [review]

**Pull request merged**